api/controllers/DashboardController.js

```diff
@@ -122,7 +132,102 @@ var dashboardController = {
             : `SELECT * FROM ${procedureToUse}(:userId, :pageSize, :offset, :sortOrder, :direction, :queryString, :secondarySortOrder, :secondaryDirection)`;
     },
 
-    _displayApplications(results, displayAppsArgs) {
+    async _displayApplications(results, displayAppsArgs) {
+        const { currentPage, req, res } = displayAppsArgs;
+        //redirect to 404 if user has manually set a page in the query string
+        if (results.length === 0) {
+            if (currentPage != 1) {
+                return res.view('404.ejs');
+            } else {
+                sails.log.error('No results found.');
+            }
+        }
+        const apiResponse = await dashboardController._getDataFromCasebook(
+            req,
+            results
+        );
+        return dashboardController._addCasebookStatusesToResults(apiResponse, {
+            ...displayAppsArgs,
+            results,
+        });
+    },
+
+    _getDataFromCasebook(req, results) {
+        // Create status retrieval request object.
+
+        // First build array of application references to be passed to the Casebook Status API for this page. Can submit up to 20 at a time.
+        const applicationReferences = results.map(
+            (resultItem) => resultItem.unique_app_id
+        );
+
+        // Create Request Structure
+
+        const leg_app_stat_struc = {
+            timestamp: Date.now().toString(),
+            applicationReference: applicationReferences,
+        };
+
+        const queryStr = apiQueryString.stringify(leg_app_stat_struc);
+
+        let certPath;
+        try {
+            certPath = req._sails.config.casebookCertificate;
+        } catch (err) {
+            sails.log.error('Null certificate path: [%s] ', err);
+            certPath = null;
```

Comment:
certPath is already `undefined`, I don't think you have to redefine it as `null` unless you're explicitly checking for null values?